### PR TITLE
[api-platform/core] Support for v3.4 and v4

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| License       | MIT
| Doc issue/PR  | n/a

* Support for the subtree split (the bundle is now available in a standalone package in addition to as part of core: `api-platform/symfony`
* Simplified configuration 

Closes https://github.com/api-platform/core/issues/6631

### CODE REVIEW

**Comment by stof** [comment]
Why is the PR title saying it is about support v4 when the recipe is for 3.4 ?

**Review by jrushlow — APPROVED** [review]
Looks good to me - merging will allow us to revert https://github.com/symfony/maker-bundle/pull/1598 over in MakerBundle

**Review by chalasr — APPROVED** [review]

**Comment by chalasr** [comment]
Ah, not sure about the failing test https://github.com/symfony/recipes/actions/runs/11029950174/job/30633515716?pr=1339#step:6:275

File: api-platform/symfony/4.0/manifest.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (maxhelias) [inline api-platform/symfony/4.0/manifest.json:1]: Shouldn't it conflict with api-platform/core < 4.0 ?

Comment (dunglas) [reply api-platform/symfony/4.0/manifest.json:1]: WDYT @soyuka?

**Comment by dunglas** [comment]
@chalasr This is because API Platform 4 requires PHP 8.2+. We forgot to bump the minimum version in `composer.json` in 4.0.0 before tagging (but the code was broken with php <8.2) and now Composer always tries to install this broken version with PHP 8.1. Maybe should we delete the 4.0.0 tag, but it will cause other kinds of issues.

For now I suggest to just merge as-is.

**Comment by dunglas** [comment]
@chalasr we finally fixed it by rewriting the tags.

**Review by chalasr — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Pull request merged**